sdk/src/main/java/com/hedera/hashgraph/sdk/RngTransaction.java

```diff
@@ -0,0 +1,91 @@
+/*-
+ *
+ * Hedera Java SDK
+ *
+ * Copyright (C) 2020 - 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.hedera.hashgraph.sdk;
+
+import com.hedera.hashgraph.sdk.proto.RandomGenerateTransactionBody;
+import com.hedera.hashgraph.sdk.proto.SchedulableTransactionBody;
+import com.hedera.hashgraph.sdk.proto.TransactionBody;
+import com.hedera.hashgraph.sdk.proto.TransactionResponse;
+import io.grpc.MethodDescriptor;
+
+import javax.annotation.Nullable;
+
+/**
+ * Random Number Generator Transaction.
+ */
+public final class RngTransaction extends com.hedera.hashgraph.sdk.Transaction<RngTransaction> {
```

Comment:
The reason we often rename things in the SDK is to provide clarity. In my opinion `RandomGenerateTransaction` does not make sense; randomly generate _what_? I'm not saying `RngTransaction` is written in stone, but _to me_ `RngTransaction` at least conveys this transaction will generate random numbers in some form or another.

Reply:
Then you should be more involved in the HIP or protobuf process to effect change there. This is the naming we decided upon and the SDKs should match it to be consistent.

Reply:
I brought this up in a previous discussion too, in that I find it difficult to use the SDK when it names things differently. It would be better to have consistency across these APIs. Not a lot of people use the HAPI directly, so in some sense the SDK is the "true" API, but these transaction names are exposed also in mirror nodes and explorers and record files and other places. People might get lost looking for an RngTransaction in the records and wonder why we couldn't have more consistency.

Reply:
We are discussing this right now, along with a few other changes to this HIP. We are proposing to use PrngTransaction as the name for it (and updating the method used in the smart contracts too)